packages/react/src/Banner/Banner.tsx

```diff
@@ -0,0 +1,323 @@
+            onClick={onDismiss}
+            className="BannerDismiss"
+            icon={XIcon}
+            variant="invisible"
+          />
+        ) : null}
+      </StyledBanner>
+    </BannerContext.Provider>
+  )
+})
+
+/**
+ * For styling, it's important that the icons and the text have the same height
+ * for alignment to occur in multi-line scenarios. Currently, we use a
+ * line-height of `20px` so that means that the height of icons should match
+ * that value.
+ */
+const StyledBanner = styled.div`
+  display: grid;
+  grid-template-columns: auto minmax(0, 1fr) auto;
+  align-items: start;
+  background-color: var(--banner-bgColor);
+  border: 1px solid var(--banner-borderColor);
+  padding: 0.5rem;
+  border-radius: ${get('radii.2')};
+
+  &[data-variant='critical'] {
+    --banner-bgColor: ${get('colors.danger.subtle')};
+    --banner-borderColor: ${get('colors.danger.muted')};
+    --banner-icon-bgColor: ${get('colors.danger.fg')};
+  }
+
+  &[data-variant='info'] {
+    --banner-bgColor: ${get('colors.accent.subtle')};
+    --banner-borderColor: ${get('colors.accent.muted')};
+    --banner-icon-bgColor: ${get('colors.accent.fg')};
+  }
+
+  &[data-variant='success'] {
+    --banner-bgColor: ${get('colors.success.subtle')};
+    --banner-borderColor: ${get('colors.success.muted')};
+    --banner-icon-bgColor: ${get('colors.success.fg')};
+  }
+
+  &[data-variant='upsell'] {
+    --banner-bgColor: ${get('colors.done.subtle')};
+    --banner-borderColor: ${get('colors.done.muted')};
+    --banner-icon-bgColor: ${get('colors.done.fg')};
+  }
+
+  &[data-variant='warning'] {
+    --banner-bgColor: ${get('colors.attention.subtle')};
+    --banner-borderColor: ${get('colors.attention.muted')};
+    --banner-icon-bgColor: ${get('colors.attention.fg')};
+  }
+
+  /* BannerIcon ------------------------------------------------------------- */
+
+  .BannerIcon {
+    display: grid;
+    place-items: center;
+    padding: 0.5rem;
+  }
+
+  .BannerIcon svg {
+    color: var(--banner-icon-bgColor);
+    fill: var(--banner-icon-bgColor);
+    height: 1.25rem;
+  }
+
+  /* BannerContainer -------------------------------------------------------- */
+
+  .BannerContainer {
+    display: grid;
+    font-size: 0.875rem;
+    align-items: start;
+    line-height: calc(20 / 14);
+    row-gap: 0.25rem;
+    column-gap: 0.25rem;
+  }
+
+  /* BannerContent ---------------------------------------------------------- */
+
+  .BannerContent {
+    display: grid;
+    row-gap: 0.25rem;
+    grid-column-start: 1;
+    margin-block: 0.5rem;
+  }
+
+  .BannerTitle {
+    margin: 0;
+    font-size: inherit;
+    font-weight: 600;
+  }
+
+  /* BannerActions ---------------------------------------------------------- */
+  .BannerActionsContainer {
+    display: flex;
+    column-gap: 0.5rem;
+    align-items: center;
+  }
+
+  .BannerActions [data-hide-on-sm] {
+    display: none;
+  }
+
+  @media screen and (min-width: 544px) {
```

Comment:
Hey @joshblack I see that you went with a media query instead of a container query. Is this due to baseline support?

We need to support something like this though, how do we do it?

![CleanShot 2024-04-12 at 08 52 45@2x](https://github.com/primer/react/assets/813754/4b30c321-1cc5-4319-b234-3327e8df2964)

Reply:
Hey @lukasoppermann! 👋 Great question, my plan is to follow up with container query support since we need to make sure media query support works as a fallback because container queries aren't available in all the browsers we target. I'll make an issue for it in the parent epic if that works for you 👍 

Sadly, like you pointed out, in the fallback case it won't correctly support those scenarios 😞 Is there anything you would like me to try out or any ideas to make sure we degrade gracefully here?

Reply:
@lukasoppermann my plan was definitely as a first pass to make sure things worked with `@media` and then add in container support as a progressive enhancement because of that baseline support 😞 

Great question for these scenarios, they seem tricky to do without any changes on the user end without `@container`. Maybe for these scenarios what we could do is add an explicit prop to override any implicit layout? Could be something as simple as "direction="column" or we could do something like a `type` or `variant` prop while we wait for our baseline to improve.

Let me know what you think!

Reply:
@lukasoppermann just wanted to provide an update, added support for a container query along with a flexbox fallback!